src/rust-crypto/RoomEncryptor.ts

```diff
@@ -83,8 +84,10 @@ export class RoomEncryptor {
      *
      * This ensures that we have a megolm session ready to use and that we have shared its key with all the devices
      * in the room.
+     *
+     * @param globalBlacklistUnverifiedDevices - True, it will not send encrypted messages to unverified devices
      */
-    public async ensureEncryptionSession(): Promise<void> {
+    public async ensureEncryptionSession(globalBlacklistUnverifiedDevices: boolean): Promise<void> {
```

Comment:
The `globalBlacklistUnverifiedDevices` variable is stored in the `RustCrypto` class https://github.com/matrix-org/matrix-js-sdk/blob/develop/src/rust-crypto/rust-crypto.ts#L91. `globalBlacklistUnverifiedDevices` can be changed during the lifetime of the the `RoomEncryptor`.

3 solutions come to my mind:
- Pass the `globalBlacklistUnverifiedDevices` variable when `RoomEncryptor#ensureEncryptionSessio` is called
- Pass `globalBlacklistUnverifiedDevices` to the `RoomEncryptor` constructor and update all the `RoomEncryptor` when `globalBlacklistUnverifiedDevices` is changing. It's a public variable so we will need to use a getter/setter to track it
- Pass a config or whatever with the `globalBlacklistUnverifiedDevices` in it to the `RoomEncryptor` and update this config when `globalBlacklistUnverifiedDevices` is changing. It's a public variable so we will need to use a getter/setter to track it

I chose the first solution, to me it is the easier more straightforward solution